examples/flutter_gallery/lib/demo/contacts_demo.dart

```diff
@@ -54,8 +54,10 @@ class _ContactItem extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final ThemeData themeData = Theme.of(context);
-    final List<Widget> columnChildren =
-        lines.sublist(0, lines.length - 1).map<Widget>((String line) => Text(line)).toList();
+    final List<Widget> columnChildren = lines
```

Comment:
Although I enjoyed jamming complex expressions like this on one line, I do think this style is generally more readable.

There are some idiomatic one-liners that aren't a clear win. This transformation seems OK:

```
List<Bar> bar = myList.map<Bar>((Foo foo) => foo.bar).toList();
```

```
List<Bar> bar = myList
  .map<Bar>((Foo foo) => foo.bar)
  .toList();
```
The same transformation applied to code within a widget, doesn't seem as useful:

```
return MyWidget(
  title: 'Bars',
  bars: myList.map<Bar>((Foo foo) => foo.bar).toList(),
);
```

Now the transformation has introduced what, at a glance, appears to be widget substructure. This isn't a win:
```
return MyWidget(
  title: 'Bars',
  bars: myList
    .map<Bar>((Foo foo) => foo.bar)
    .toList(),
);
```

If the formatter could recognize `myList.map<Bar>((Foo foo) => foo.bar).toList(),` as legitimate alternative way to format the expression, it could leave the hand formatted one-liner alone in this case.

Reply:
as long as myList.map<Bar>((Foo foo) => foo.bar).toList() is less than 80 chars that shouldn't be a problem. 
The preferred style could also be tweaked depending on if you are already inside a list of named arguments.